lib/components/Forms/Fields/NumberInput/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { forwardRef, useEffect, useState } from "react"
+import { Input, InputProps } from "../Input"
+
+const FORMAT = /^([0-9]+)$/
+// eslint-disable-next-line no-useless-escape
+const FORMAT_WITH_DECIMALS = /^([0-9]+)(?:[\.,]([0-9]+)?)?$/
+
+const parseValue = (value: string) => parseFloat(value.replace(",", "."))
+
+type NumberInputProps = Omit<InputProps, "value" | "type" | "onChange"> & {
+  locale: string
+  value?: number
+  maxDecimals?: number
+  onChange?: (value: number | null) => void
+}
+
+export const NumberInput = forwardRef<HTMLInputElement, NumberInputProps>(
+  ({ locale, value, maxDecimals, onChange, ...props }, ref) => {
+    const formatValue = (value: number, maxDecimals?: number) =>
+      new Intl.NumberFormat(locale, {
+        maximumFractionDigits: maxDecimals,
+        useGrouping: false,
+      }).format(value)
```

Comment:
This function seems a utility, I will move it out (adding the locale as a param)